src/Funogram.Telegram/Bot.fs

```diff
@@ -92,8 +92,13 @@ let cmdScan (format: PrintfFormat<_, _, _, _, 't>) (handler: 't -> UpdateContext
   |> not
 
 let private runBot config me updateArrived updatesArrived =
-  let bot data = api config data
-
+  let bot data =
+    async{
+    try
+      return! api config data
+    with ex ->
+        return Error { ErrorCode = 0; Description = ex.Message }
```

Comment:
Why is it needed?

Normally, and exceptions will be caught by the `try…catch` block in the `loopAsync`, and delivered to the caller via `config.OnError`.

In this code, you are losing the stack trace and any details such as inner exceptions.

Could you please clarify the problem you are solving here, so we may think about other possible solutions?

Reply:
We had a discussion with @vshapenko about this PR. This mr is proposed to correctly handle HTTP errors.

At this moment we have no filter for exception and in case of ANY exception we will try to increment update's offset:
```fsharp
with ex ->
    config.OnError ex
    return! loopAsync (offset + 1L)
```

This logic may be correct on response parsing errors but it is not correct in case of any HTTP errors (for example 504) or in case of network problem (for example) on client side (bot's side).

Actually I have bots with several years uptime and I have never encountered such errors on tg severs. Anyway I have created another one PR #56 for proper handling HTTP errors. Please, take a look.

Reply:
I still am not found of `"HTTP_ERROR"` constant, but overall, #56 I like better. Thanks foe explanation, folks!